src/firebase/api/users.js

```diff
@@ -173,3 +177,113 @@ export const updateUserData = async (updatedData) => {
     };
   }
 };
+
+/**
+ * Sign in user.
+ * @param {String} email email of user.
+ * @param {String} password password of user.
+ * @return {user} authenticated user object.
+ */
+export const signIn = async (email, password) => {
+  try {
+    const userAuth = await auth
+      .signInWithEmailAndPassword(email, password);
+    return {
+      status: StatusCodes.OK,
+      data: userAuth,
+      error: null,
+    };
+  } catch (error) {
+    if (error.code === 'auth/wrong-password' ||
+        error.code === 'auth/user-not-found') {
+      return {
+        status: StatusCodes.NOT_FOUND,
+        data: null,
+        error,
+      };
+    }
+    return {
+      status: StatusCodes.INTERNAL_SERVER_ERROR,
+      data: null,
+      error,
+    };
+  }
+};
+
+/**
+ * Sign current logged in user out.
+ * @return {Object} response object.
+ */
+export const signOut = async () => {
+  try {
+    await auth.signOut();
+    return {
+      status: StatusCodes.OK,
+      data: userAuth,
+      error: null,
+    };
+  } catch (error) {
+    return {
+      status: StatusCodes.INTERNAL_SERVER_ERROR,
+      data: null,
+      error,
+    };
+  }
+};
+
+/**
+ * This function is for users who do not what to sign in.
+ * Call this function to sign in user anonymously.
+ * @param {User} userData user's data
+ * @return {Object} user auth or error object.
+ */
+export const signInAnonymousUser = async (userData) => {
+  try {
+    const userAuth = await auth.signInAnonymously();
+    await generateUserDoc(userAuth.user, userData);
+    return {
+      status: StatusCodes.CREATED,
+      data: userAuth,
+      error,
+    };
+  } catch (error) {
+    return {
+      status: StatusCodes.INTERNAL_SERVER_ERROR,
+      data: null,
+      error,
+    };
+  }
+};
+
+/**
+ * Link user to their current anonymousAccount.
+ * @param {String} email email of user.
+ * @param {String} password password of user.
+ * @return {Object} userAuth or error object.
+ */
+export const linkUsetToAnonymousAccount = async (email, password) => {
+  try {
+    const credential = auth.EmailAuthProvider.credential(email, password);
+    const userAuth = await auth.currentUser.linkWithCredential(credential);
+    return {
+      status: StatusCodes.OK,
+      data: userAuth,
+      error: 'Linking success',
```

Comment:
Shall we use 'message' or 'error' for this message? If so we can unify the return format. {status, data, error}

Reply:
let's add this as a ticket for the next sprint to unify all responses.